repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/samlpolicy/SamlAttributeMappingTest.groovy

```diff
@@ -414,15 +422,23 @@ class SamlAttributeMappingTest extends ReposeValveTest {
         !json.access.'RAX-AUTH:extendedAttributes'
     }
 
-    def "an extended attribute with no value is added to the request/response when added by the policy"() {
+    @Unroll
+    def "an extended attribute with no value is added to the request/response when added by the #contentType policy"() {
         given: "a mapping policy an extended attribute with no value"
         def extUserAttribName = "foo"
         def extUserAttribValues = ["{0}"]
         def remotePath = "()"
         def remoteValue = [[path: remotePath]]
-        def mappingPolicy = createMappingJsonWithValues(
-            userExtAttribs: [(extUserAttribName): extUserAttribValues],
-            remote: remoteValue)
+        def mappingPolicy = contentType == APPLICATION_XML ?
+            createMappingJsonWithValues(
+                userExtAttribs: [(extUserAttribName): extUserAttribValues],
+                remote: remoteValue) : contentType == APPLICATION_JSON ?
+            createMappingJsonWithValues(
+                userExtAttribs: [(extUserAttribName): extUserAttribValues],
+                remote: remoteValue) :
+            createMappingYamlWithValues(
+                userExtAttribs: [(extUserAttribName): extUserAttribValues],
+                remote: remoteValue)
```

Comment:
At this point, I feel like using a standard try/catch block would make the intent and behavior clearer, but if you do that the standard Java way, you end up with some assignment repetition:
```
def mappingPolicy
if (contentType == APPLICATION_XML) {
    mappingPolicy = createMappingXmlWithValues(
                userExtAttribs: [(extUserAttribName): extUserAttribValues],
                remote: remoteValue)
} else if (contentType == APLICATION_JSON) {
    mappingPolicy = createMappingJsonWithValues(
                userExtAttribs: [(extUserAttribName): extUserAttribValues],
                remote: remoteValue)
} else {
    mappingPolicy = createMappingYamlWithValues(
                userExtAttribs: [(extUserAttribName): extUserAttribValues],
                remote: remoteValue)
}
```

Instead, you could use a code block, but Groovy doesn't make that super easy since code blocks and closures share syntax. So... you might actually have to use a closure or a hacky construct:
```
def mapingPolicy = {
    if (contentType == APPLICATION_XML) {
        createMappingXmlWithValues(
            userExtAttribs: [(extUserAttribName): extUserAttribValues],
            remote: remoteValue)
    } else if (contentType == APLICATION_JSON) {
        createMappingJsonWithValues(
            userExtAttribs: [(extUserAttribName): extUserAttribValues],
            remote: remoteValue)
    } else {
        createMappingYamlWithValues(
            userExtAttribs: [(extUserAttribName): extUserAttribValues],
            remote: remoteValue)
    }
}()
```

Reply:
Completely agree with extracting this to a method for reuse.